torch/_inductor/compile_fx.py

```diff
@@ -193,7 +193,10 @@ def wrapper(gm: torch.fx.GraphModule, example_inputs: List[torch.Tensor], **kwar
                     "cpp_wrapper": False,
                     "cudagraphs": False,
                 }
-                compiled = inner_compile(gm, example_inputs, **kwargs_patched)
+                # deepcopy(gm) makes sure no graph modification from the first pass will
+                # leak to the second pass. It does increase memory pressure, but the problem
+                # can be alleviated once we have FakeModule.
+                compiled = inner_compile(deepcopy(gm), example_inputs, **kwargs_patched)
```

Comment:
Let's not use deepcopy here since it might try to copy non-FX stuff referenced by the graph.

I think you can just run torch.fx.Transformer() to shallow copy an FX graph.